src/Symfony/Component/HttpClient/Internal/CurlClientState.php

```diff
@@ -67,8 +67,10 @@ public function __construct(int $maxHostConnections, int $maxPendingPushes)
             return;
         }
 
-        curl_multi_setopt($this->handle, \CURLMOPT_PUSHFUNCTION, function ($parent, $pushed, array $requestHeaders) use ($maxPendingPushes) {
-            return $this->handlePush($parent, $pushed, $requestHeaders, $maxPendingPushes);
+        // Passing a non-static closure can lead to a segfault
+        $multi = $this;
+        curl_multi_setopt($mh, \CURLMOPT_PUSHFUNCTION, static function ($multi, $parent, $pushed, array $requestHeaders) use ($maxPendingPushes) {
```

Comment:
is `$multi` really passed as the first argument of the push function by curl ? this looks suspicious to me.

Reply:
fixed